### DISCUSSION TITLE
Weekly Update - 2021-09-16

### DISCUSSION THREAD

**Discussion by matklad** [near/nearcore#4832] [Contract Runtime]
Done:

* New gas counter correctly enforces the `max_gas_burnt` limit. 
* Upgraded Rust version used to compile nearcore
* Estimated wasm instruction cost with the new gas counter, got 4x reduction. That's big, but not as good as our original estimates. The original estimates were wrong, as they didn't take LTO into account. 

In Progress:

* Preparing new `gas_counter` for rollout. Sadly, we discovered that implementation details about our current stack and gas metering leak to the protocol in corner-cases.
* Preparing to merge wasmer2 into nearcore: published the crates

**Comment by maxzaver** [comment]
@matklad @ailisp @olonho Can we increase or decrease this variable [here](https://github.com/near/nearcore/blob/06d05bce3fab578a91b210d7354fb17427fcc9d6/nearcore/res/runtime_configs/116.json#L155) to make sure all past transactions succeed/fail in exactly the same way with new gas counter?

**Comment by maxzaver** [comment]
@Longarithm FYI Wasmer 2 will likely have different module loading costs.
